src/components/input.js

```diff
@@ -199,114 +199,64 @@ export const ValidatedInput = props => {
-        suggestionHighlighted: { backgroundColor: colors.light(0.4) }
-      }
-    })
-  }
-}
-
-export class AutocompleteSearch extends Component {
-  static propTypes = {
-    value: PropTypes.string.isRequired,
-    onChange: PropTypes.func.isRequired,
-    onSuggestionSelected: PropTypes.func.isRequired,
-    suggestions: PropTypes.array,
-    renderSuggestion: PropTypes.func,
-    theme: PropTypes.object
-  }
-
-  static defaultProps = {
-    suggestions: [],
-    renderSuggestion: _.identity
-  }
-
-  constructor(props) {
-    super(props)
-    this.state = { show: false }
-    this.id = props.id || _.uniqueId('AutocompleteSearch_')
-  }
-
-  render() {
-    const { value, onChange, onSuggestionSelected, suggestions, renderSuggestion, theme, ...props } = this.props
-    const { show } = this.state
-    return h(Autosuggest, {
-      id: this.id,
-      inputProps: { id: this.id, value, onChange: onChange ? (e => onChange(e.target.value)) : undefined, ...props },
-      suggestions: show ? (value ? [value, ..._.filter(Utils.textMatch(value), suggestions)] : suggestions) : [],
-      onSuggestionsFetchRequested: () => this.setState({ show: true }),
-      onSuggestionsClearRequested: () => this.setState({ show: false }),
-      onSuggestionSelected: (e, { suggestionValue }) => onSuggestionSelected(suggestionValue),
-      getSuggestionValue: _.identity,
-      shouldRenderSuggestions: value => value.trim().length > 0,
-      renderSuggestionsContainer: ({ containerProps, children }) => {
-        return children && h(AutocompleteSuggestions, { containerProps, children, target: this.id })
-      },
-      renderSuggestion,
-      renderInputComponent: inputProps => h(TextInput, { nativeOnChange: true, ...inputProps }),
-      theme: _.merge({
-        container: { width: '100%' },
-        suggestionsList: { margin: 0, padding: 0 },
-        suggestion: styles.suggestion,
-        suggestionHighlighted: { backgroundColor: colors.light(0.4) }
-      }, theme)
-    })
-  }
+  return h(Downshift, {
+    selectedItem: value,
+    onInputValueChange: onChange,
+    inputId: id
+  }, [
+    ({ getInputProps, getMenuProps, getItemProps, isOpen, openMenu, toggleMenu, highlightedIndex }) => {
+      return div({ style: { width: '100%' } }, [
+        h(TextInput, getInputProps({
+          style,
+          type: 'search',
+          onFocus: openOnFocus ? openMenu : undefined,
+          onKeyDown: e => {
+            if (e.key === 'Escape') {
+              (value || isOpen) && e.stopPropagation() // prevent e.g. closing a modal
+              if (!value || isOpen) { // don't clear if blank (prevent e.g. undefined -> '') or if menu is shown
+                e.nativeEvent.preventDownshiftDefault = true
+                e.preventDefault()
+              }
+              toggleMenu()
+            }
+          },
+          nativeOnChange: true,
+          ...props
+        })),
+        h(AutocompleteSuggestions, {
```

Comment:
Ah, I had gone with unconditional rendering because https://github.com/downshift-js/downshift#getmenuprops says
> Note that for accessibility reasons it's best if you always render this element whether or not downshift is in an `isOpen` state.

I'll switch back to conditional rendering, though, since none of these are for places where custom text isn't possible.